### PR TITLE
Deprecated nonkeyword arguments for set_codes function

### PR DESCRIPTION
**Pull request by Pydare**
- [ ] xref #41485
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
this is failing some precommit tests, pls rebase and fix up.

**Review by simonjayhawkins — COMMENTED** [review]
@Pydare merge conflicts to resolve and whitespace issues, need to run `black` or `pre-commit`

https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -677,6 +677,7 @@ Deprecations
 - Deprecated setting :attr:`Categorical._codes`, create a new :class:`Categorical` with the desired codes instead (:issue:`40606`)
 - Deprecated behavior of :meth:`DatetimeIndex.union` with mixed timezones; in a future version both will be cast to UTC instead of object dtype (:issue:`39328`)
 - Deprecated using ``usecols`` with out of bounds indices for ``read_csv`` with ``engine="c"`` (:issue:`25623`)
+- Deprecated passing arguments as positional (except for ``"codes"``) in :meth:`MultiIndex.codes` (:issue:`41485`)
```
[the remark this answers is not in the document]
Comment (MarcoGorelli) [reply doc/source/whatsnew/v1.3.0.rst:682]: the other whats new notes for this issue use two backticks and then quotes - I'll fix up them all into a single whatsnew message at the end

**Review by simonjayhawkins — APPROVED** [review]
Thanks @Pydare lgtm pending green

**Comment by jreback** [comment]
thanks @Pydare

**Pull request merged**